### PR TITLE
Update Flatbuffer Schema

### PR DESCRIPTION
**Pull request by rdspring1**
This PR updates the Flatbuffer schema in preparation for serializing the rest of the stack.

Linked follow-up PR: https://github.com/NVIDIA/Fuser/pull/505

The new schema handles:
1. FusionExecutorCache
2. FusionKernelRuntime ---> We use the original input ArgAbstractHolder to schedule the Fusion.
3. FusionExecutor
4. ExecutorEntry ---> We store an entry for each input used with the kernel.
5. GlobalBufferInfo
6. LaunchParams
7. KernelArgumentHolder
8. ArgAbstract ---> Bool, ComplexDouble, Double, Long, Philox, ScalarCpu, TensorArgAbstract
9. InputsIdLookup

### CODE REVIEW

**Comment by rdspring1** [comment]
!build

**Comment by zasdfgbnm** [comment]
Do we have test for this PR? I am asking because, in the future, I will add a new `ArgAbstract` for TMA tensor map. If I forget to update the schema for this new argument type, will I silently break serde, or will the CI yell at me?

**Comment by rdspring1** [comment]
This PR only updates the schema in preparation for another PR I'm about to open. The next PR will have all the tests.

I'm using the `ArgAbstract` objects to scheduler the `FusionKernelRuntime`.
 * `FusionKernelRuntime(std::unique_ptr<Fusion> fusion, const KernelArgumentHolder& inputs, ...)`
Will the `TMA tensor map` be used in this ^^^ context?

**Comment by zasdfgbnm** [comment]
I am not familiar with how `FusionKernelRuntime` is used, but it should be similar to `PhiloxCudaState`, which is computed by `FusionExecutor` and passed to the kernel as arguments.

**Comment by rdspring1** [comment]
!build

File: csrc/serde/fusion_cache.fbs [hunk 1]
```diff
@@ -188,8 +206,91 @@ table ScalarInput {
...
+  cache_id : ulong;
+}
+
 //
 // =====================================================================================
+// Tables for LaunchParams, GlobalBufferInfo, ExecutorEntry used in FusionExecutor
+
+// This table holds the cached launch parameters for a kernel.
+table LaunchParams {
+  gdimx : long;
```
Comment (jjsjann123) [inline csrc/serde/fusion_cache.fbs:255]: do we need long here? I thought int would be enough... https://docs.nvidia.com/cuda/cuda-c-programming-guide/index.html#features-and-technical-specifications

Comment (jjsjann123) [reply csrc/serde/fusion_cache.fbs:255]: oh, NVM, I realized that this is merely a mapping to what we have in LaunchParams defined in executor_params.h.

**Review by jjsjann123 — COMMENTED** [review]
Sorry for the delayed review, I wasn't on top of this PR.
Looks like this is just a mechanical change to add the schema. Do we want to have this one merged now as-is and patch it in the later PR that enables the function?

**Comment by rdspring1** [comment]
> Do we want to have this one merged now as-is and patch it in the later PR that enables the function?

I planned to merge this PR first to make https://github.com/NVIDIA/Fuser/pull/505 easier to review.

**Review by jjsjann123 — APPROVED** [review]
stamping for mechanical changes. We'll do the real review in follow up PR.

**Comment by rdspring1** [comment]
!build

**Comment by rdspring1** [comment]
!build

**Comment by rdspring1** [comment]
!build

**Pull request merged**